forem/forem
# Crayons: Pills component

## What type of PR is this? (check all applicable)


## Description

This PR introduces new UI component, called Pills. I'll copy'n'paste some info that has already been added to the component documentation itself because I think it explains it pretty well:

> Pill is an element representing a relatively short form of data, i.e. a short label. Pill would visually encapsulate such a data in... well, a _pill_.
> 
> Generally pills should occur in a herd 🐺. So it's type of data that you would expect to see as set of values. For example, imagine a list of roles associated with some user's account. This could be represented as bunch of pills (of course depending on the context) - each pill being one role.
> 
> Pills could be used to simply represent some data but they can also have an action (only one action per pill) baked-in.

That's in general a description for this component. Please read Pills.mdx file from Storybook to have more context.

I've turned this UI element into a component because this is literally something I needed for Admin Member View I'm currently working on. And also this is something that we will likely use more often on other views, especially in Admin world.

## Related Tickets & Documents

- [This is where I will use it initially](https://github.com/forem/forem/pull/16174).
- [Design](https://www.figma.com/file/gHpwugeu9jFGW4d9u3ReuR/Crayons?node-id=436%3A212)

## QA Instructions, Screenshots, Recordings

Spin up `yarn storybook` and go to **Pills** section. This will only appear in Storybook as it hasn't been used anywhere in product yet.

### UI accessibility concerns?

Nope.

## Added/updated tests?

Nope.

## [Forem core team only] How will this change be communicated?

1. Storybook
2. Forem.team post + individual changelog post, also n forem.team (and likely on forem.dev as well)
